Sources/SwiftSyntax/Trivia.swift

```diff
@@ -42,6 +42,101 @@ public struct Trivia: Sendable {
     pieces.isEmpty
   }
 
+  /// The string contents of all the comment pieces with any comments tokens trimmed.
+  public var commentValue: String {
+    var comments = [Substring]()
+
+    // Determine if all line comments have a single space
+    lazy var allLineCommentsHaveSpace: Bool = {
+      return pieces.allSatisfy { piece in
+        switch piece {
+        case .lineComment(let text):
+          return text.hasPrefix("// ")
+        case .docLineComment(let text):
+          return text.hasPrefix("/// ")
+        default:
+          return true
+        }
+      }
+    }()
+
+    // Returns a substring with leading and trailing spaces removed.
+    func trimWhitespace(_ text: Substring) -> Substring {
+        let trimmed = text.drop(while: { $0 == " " })
+        let reversed = trimmed.reversed()
+        let trimmedEnd = reversed.drop(while: { $0 == " " })
+        let final = trimmedEnd.reversed()
+        return Substring(final)
+    }
+
+    // Strips /* */ markers and aligns content by removing common indentation.
+    func processBlockComment(_ text: Substring) -> String {
+      var lines = text.split(separator: "\n", omittingEmptySubsequences: false)
+
+      let minIndentation =
+        lines
+        .dropFirst()
+        .filter { !$0.isEmpty }
+        .map { $0.prefix { $0 == " " }.count }
+        .min() ?? 0
+
+      var firstLineRemoved = false;
+      var firstLine = lines[0]
+      if trimWhitespace(firstLine) == "/*" || trimWhitespace(firstLine) == "/**" {
+          lines.removeFirst()
+          firstLineRemoved = true;
+      } else {
+          firstLine = firstLine.hasPrefix("/**") ? firstLine.dropFirst(3) : firstLine.dropFirst(2)
+          while firstLine.first?.isWhitespace == true {
+              firstLine = firstLine.dropFirst()
+          }
+          lines[0] = firstLine
+      }
+      
+      if let lastLine = lines.last {
+          if trimWhitespace(lastLine) == "*/" {
```

Comment:
Similar here, would it be easier to do normal processing and drop the last line from `unindentedLines` if it is whitespace-only?